src/connector.cpp

```diff
@@ -2435,14 +2447,29 @@ void HostConnector::hostEnsureStereoChain(const uint8_t row, const uint8_t block
     // ----------------------------------------------------------------------------------------------------------------
     // part 1: deal with dual-mono and disconnect blocks where needed
 
+    bool sideChainToBeUpdated = false;
+
     for (uint8_t bl = blockStart; bl < NUM_BLOCKS_PER_PRESET; ++bl)
     {
         const Block& blockdata(chain.blocks[bl]);
         if (isNullBlock(blockdata))
             continue;
 
         const bool oldDualmono = _mapper.get(_current.preset, row, bl).pair != kMaxHostInstances;
-        const bool newDualmono = previousPluginStereoOut && blockdata.meta.numInputs == 1;
+        // activate dual mono if previous plugin is stereo or also dualmono
```

Comment:
this logic got a bit more complicated now and it is present in 2 places, it should become a function of its own.